server/tasks/import_git_repo.py

```diff
@@ -0,0 +1,130 @@
+#!/usr/bin/env python
+# -*- coding: utf-8 -*-
+
+import git
+import json
+import sys
+import time
+import traceback
+from girder.models.folder import Folder
+from girder.models.token import Token
+from girder.models.user import User
+from girder.utility import JsonEncoder
+from girder.plugins.jobs.constants import JobStatus
+from girder.plugins.jobs.models.job import Job
+
+from ..constants import InstanceStatus, TaleStatus
+from ..models.instance import Instance
+from ..models.tale import Tale
+
+
+def run(job):
+    jobModel = Job()
+    jobModel.updateJob(job, status=JobStatus.RUNNING)
+
+    url, = job["args"]
+    if "@" in url:
+        repo_url, branch = url.split("@")
+    else:
+        repo_url = url
+        branch = "master"
+
+    user = User().load(job["userId"], force=True)
+    tale = Tale().load(job["kwargs"]["taleId"], user=user)
+    spawn = job["kwargs"]["spawn"]
+    change_status = job["kwargs"].get("change_status", True)
+    token = Token().createToken(user=user, days=0.5)
+
+    progressTotal = 1 + int(spawn)
+    progressCurrent = 0
+
+    try:
+        # 1. Checkout the git repo
+        jobModel.updateJob(
+            job,
+            status=JobStatus.RUNNING,
+            progressTotal=progressTotal,
+            progressCurrent=progressCurrent,
+            progressMessage="Cloning the git repo",
+        )
+
+        workspace = Folder().load(tale["workspaceId"], force=True)
+        try:
+            repo = git.Repo.init(workspace["fsPath"])
+            origin = repo.create_remote("origin", repo_url)
+            origin.fetch()
```

Comment:
Catching exception on fetch would allow for check of whether specified repo exists/was accessible. I expect fetch fails on bad URL and private repos.  Also the returned FetchInfo has the default branch name.

Reply:
I believe we catch any exception right now so it should show up. Unless I misunderstood your suggestion.

Reply:
Sorry, unclear as usual.  I was thinking about handling the different exceptions for `fetch` and `checkout` when deciding whether to remove the `.git` directory.  Fetch errors should rollback (invalid URL or private repo).  Checkout errors might not rollback (can't checkout because workspace has a file with the same name as the remote repo).  Now that you're removing `.git` in both cases I think your approach is better than leaving the user in a half-checked-out state.